packages/@aws-cdk/aws-ecs/lib/container-definition.ts

```diff
@@ -478,7 +478,7 @@ function getHealthCheckCommand(hc: HealthCheck): string[] {
     return hcCommand;
   }
 
-  if (cmd[0] !== "CMD" || cmd[0] !== 'CMD-SHELL') {
+  if (cmd[0] !== "CMD" && cmd[0] !== 'CMD-SHELL') {
```

Comment:
Can you add a unit test to ensure that this check is working as expected?

Reply:
@pkandasamy91 the [existing test ](https://github.com/awslabs/aws-cdk/pull/2462/files#diff-053089ea576f1ea4920bd44e82d220cbR363) should be sufficient.

Reply:
Will add missing tests in a separate PR.